overlord/configstate/config/helpers.go

```diff
@@ -98,7 +100,9 @@ func GetFromChange(snapName string, subkeys []string, pos int, config map[string
 		if !ok {
 			raw = jsonRaw(value)
 		}
-		err := json.Unmarshal([]byte(*raw), result)
+		dec := json.NewDecoder(bytes.NewReader(*raw))
```

Comment:
pardon my ignorance please, but why is this not using util.DecodeJsonWithNumbers() ? Maybe a small comment with the reason behind this?